overlord/state/state.go

```diff
@@ -171,6 +172,8 @@ type marshalledState struct {
 	LastTaskId   int `json:"last-task-id"`
 	LastLaneId   int `json:"last-lane-id"`
 	LastNoticeId int `json:"last-notice-id"`
+
+	NoticeLastDate time.Time `json:"notice-last-date,omitempty"`
```

Comment:
Is there any way to get json to marshal this using a resolution more precise than 1 second? Perhaps a custom {,un}marshaller which uses `RFC3339Nano` would be best?

Reply:
In particular, what if this were a string...

Reply:
Go's `encoding/json` does use nanosecond precision for these: https://go.dev/play/p/nmJ_2l5G6QI

Reply:
Ah okay. Then I think the test strings should use nanosecond precision as well, right? I commented on the locations.

Reply:
As commented below, it seems that the trailing zeros are removed, so for the tests to pass, they must not have decimals.